packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js

```diff
@@ -498,7 +498,15 @@ module.exports.external = [
           );
           partial = handlebarsInstance.partials[templatePath];
         }
-        return JSON.parse(jsonProcessor.Process(partial(inObj.hash)));
+        var result = partial(inObj.hash);
+        var processedResult = JSON.parse(jsonProcessor.Process(result));
+
+        // Check if the processedResult is undefined or an empty object
+        if (processedResult === undefined || (Object.keys(processedResult).length === 0 && processedResult.constructor === Object)) {
```

Comment:
evaluate is used in with blocks. Issue is handlebars partials, if they return nothing, still return empty objects. So when you do 
{{#with (evaluate func obj)}}
stuff
otherstuff
{{/with}}

you expect the stuff inside the with to not get evaluated for undefined return types, since thats the way with is supposed to work. The issue is evaluate is returning empty objects, and so then we do evaluate the inside of the with but pass empty objects inside it, and it causes errors.